pypiserver/core.py

```diff
@@ -222,7 +222,7 @@ def __init__(self, pkgname, version, fn=None, root=None, relfn=None, replaces=No
     def __repr__(self):
         return "%s(%s)" % (
             self.__class__.__name__,
-            ", ".join(["%s=%r" % (k, getattr(self, k))
+            ", ".join(["%s=%r" % (k, getattr(self, k, 'AttributeError'))
```

Comment:
Added this just to fix a `pytest` error when two `PkgFile`s didn't match - `pytest` was unable to show the diff, because artificially instantiated `PkgFile`s in tests didn't have the `_fname_and_hash` attribute. Instead of an object diff, I was getting:
```
tests/test_manage.py:216: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../.pyenv/versions/3.8.0/lib/python3.8/unittest/mock.py:919: in assert_called_once_with
    return self.assert_called_with(*args, **kwargs)
../../../.pyenv/versions/3.8.0/lib/python3.8/unittest/mock.py:907: in assert_called_with
    raise AssertionError(_error_message()) from cause
../../../.pyenv/versions/3.8.0/lib/python3.8/unittest/mock.py:901: in _error_message
    msg = self._format_mock_failure_message(args, kwargs)
../../../.pyenv/versions/3.8.0/lib/python3.8/unittest/mock.py:799: in _format_mock_failure_message
    expected_string = self._format_mock_call_signature(args, kwargs)
../../../.pyenv/versions/3.8.0/lib/python3.8/unittest/mock.py:794: in _format_mock_call_signature
    return _format_call_signature(name, args, kwargs)
../../../.pyenv/versions/3.8.0/lib/python3.8/unittest/mock.py:2333: in _format_call_signature
    args_string = ', '.join([repr(arg) for arg in args])
../../../.pyenv/versions/3.8.0/lib/python3.8/unittest/mock.py:2333: in <listcomp>
    args_string = ', '.join([repr(arg) for arg in args])
pypiserver/core.py:225: in __repr__
    ", ".join(["%s=%r" % (k, getattr(self, k))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.0 = <list_iterator object at 0x7f2cfb03ab80>

>   ", ".join(["%s=%r" % (k, getattr(self, k))
                          for k in sorted(self.__slots__)]))
E                         AttributeError: _fname_and_hash
```